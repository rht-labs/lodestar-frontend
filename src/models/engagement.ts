export interface Engagement {
  name: string;
}
export class Engagement {
  constructor({ name }: Engagement) {
    this.name = name;
  }
  name: string;
}
