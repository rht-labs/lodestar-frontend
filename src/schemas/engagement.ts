import faker from 'faker/locale/en_US';
import { LaunchData } from './launch_data';
import { GitCommit } from './git_commit';
import { CreationDetails } from './creation_details';
import { ClusterStatus } from './cluster_status';
export interface Engagement {
  additional_details?: string;
  archive_date: Date;
  commits: GitCommit[];
  customer_contact_email: string;
  customer_contact_name: string;
  customer_name: string;
  description: string;
  end_date: Date;
  engagement_lead_email: string;
  engagement_lead_name: string;
  engagement_users: Array<any>;
  last_update: string;
  location: string;
  mongo_id?: string;
  ocp_cloud_provider_name: string;
  ocp_cloud_provider_region: string;
  ocp_cluster_size: string;
  ocp_persistent_storage_size: string;
  ocp_sub_domain: string;
  ocp_version: string;
  project_id: number;
  project_name: string;
  start_date: Date;
  technical_lead_email: string;
  technical_lead_name: string;
  launch?: LaunchData;
  creation_details: CreationDetails;
  last_update_by_name: string;
  suggested_subdomain?: string;
  status: ClusterStatus;
}
export abstract class Engagement {
  static fromFake(staticData = false): Engagement {
    return {
      additional_details: staticData
        ? 'Additional information here'
        : faker.lorem.paragraphs(2),
      archive_date: staticData ? new Date(2020, 6, 30) : faker.date.recent(),
      commits: [],
      customer_contact_email: staticData
        ? 'bob@doe.com'
        : faker.internet.email(),
      customer_contact_name: staticData
        ? 'Bob Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      customer_name: staticData ? 'NASA' : faker.company.companyName(),
      description: staticData
        ? "It's rocket science"
        : faker.lorem.paragraphs(5),
      end_date: staticData ? new Date(2020, 6, 1) : faker.date.future(),
      engagement_users: staticData ? [] : [],
      engagement_lead_email: staticData
        ? 'alice@doe.com'
        : faker.internet.email(),
      engagement_lead_name: staticData
        ? 'Alice Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      last_update: staticData ? '2020-01-01' : new Date().toString(),
      location: staticData
        ? 'Nashville, TN'
        : `${faker.address.city()}, ${faker.address.stateAbbr()}`,
      mongo_id: '1',
      ocp_cloud_provider_name: staticData ? 'AWS' : 'AWS',
      ocp_cloud_provider_region: staticData
        ? 'N. Virginia'
        : faker.lorem.word(),
      ocp_cluster_size: staticData ? 'Large' : 'Large',
      ocp_persistent_storage_size: staticData
        ? '100G'
        : `${faker.random.number()}G`,
      ocp_sub_domain: staticData ? 'ordoabchao' : faker.lorem.word(),
      ocp_version: staticData ? '4.4' : faker.random.number().toString(),
      project_id: staticData ? 1 : faker.random.number(),
      project_name: staticData ? 'Boots on the Moon' : faker.company.bsNoun(),
      start_date: staticData ? new Date(2020, 5, 1) : faker.date.recent(),
      technical_lead_email: staticData ? 'eve@doe.com' : faker.internet.email(),
      technical_lead_name: staticData
        ? 'Eve Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      creation_details: staticData
        ? {
            created_by_email: 'dwasinger@redhat.com',
            created_by_user: 'dwasinge',
            created_on: new Date(2020, 1, 1),
          }
        : {
            created_by_email: faker.internet.email(),
            created_by_user: `${faker.name.firstName()} ${faker.name.lastName()}`,
            created_on: faker.date.recent(),
          },
      last_update_by_name: staticData
        ? 'James Doe'
        : `${faker.name.firstName()} ${faker.name.lastName()}`,
      launch: staticData
        ? null
        : faker.random.boolean()
        ? {
            launched_by: faker.name.firstName(),
            launched_date_time: faker.date.recent(),
          }
        : null,
      status: ClusterStatus.fromFake(staticData),
    };
  }
}

export enum EngagementStatus {
  active = 'active',
  past = 'past',
  upcoming = 'upcoming',
}

export const getEngagementStatus = (
  engagement: Engagement
): EngagementStatus => {
  if (!engagement) {
    return null;
  }
  const Today = new Date();
  const { launch, end_date } = engagement;
  const hasLaunched = !!launch?.launched_date_time;
  if (hasLaunched && end_date >= Today) {
    return EngagementStatus.active;
  } else if (hasLaunched && end_date < Today) {
    return EngagementStatus.past;
  } else {
    return EngagementStatus.upcoming;
  }
};
