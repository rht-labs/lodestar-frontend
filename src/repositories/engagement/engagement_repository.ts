import { Engagement } from '../../models/engagement';

export abstract class EngagementRepository {
  abstract async fetchEngagements(): Promise<Engagement[]>;
}
