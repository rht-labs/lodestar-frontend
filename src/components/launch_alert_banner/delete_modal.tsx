import React from 'react';
import { Modal, ModalVariant, Button } from '@patternfly/react-core';
import { useModalVisibility } from '../../context/edit_modal_visibility_context/edit_modal_visibility_hook';
import {useEngagements} from "../../context/engagement_context/engagement_hook";
import {AnalyticsCategory} from "../../schemas/analytics";
import {useAnalytics} from "../../context/analytics_context/analytics_context";
import {Engagement} from "../../schemas/engagement";
import {useHistory} from "react-router";

export interface DeleteModalProps {
  isOpen: boolean;
  engagement: Engagement
}
export function DeleteModal(
  props: DeleteModalProps
) {
  const {
    deleteEngagement
  } = useEngagements();

  const history = useHistory();
  const { logEvent } = useAnalytics();

  const deleteThisEngagement = async () => {
    history.push({
      pathname: `/app/engagements/`,
    });
    await deleteEngagement(props.engagement);
    logEvent({
      action: 'Delete engagement',
      category: AnalyticsCategory.engagements,
    });
  };

  const { requestClose } = useModalVisibility();
  return (
    <Modal
      variant={ModalVariant.small}
      title={`Delete ${props.engagement?.project_name}?`}
      isOpen={props.isOpen}
      onClose={requestClose}
      actions={[
        <Button key="delete" variant="danger" onClick={deleteThisEngagement}>
          Delete
        </Button>,
        <Button key="cancel" variant="secondary" onClick={requestClose}>
          Cancel
        </Button>
      ]}
    >
     Are you sure you want to delete {props.engagement?.customer_name} ({props.engagement?.project_name})? By deleting this engagement, all related data will be lost and can not be retrieved.
    </Modal>
  );
}
