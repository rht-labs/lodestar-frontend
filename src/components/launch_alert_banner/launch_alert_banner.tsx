import React from 'react';
import { Engagement } from '../../schemas/engagement';
import { Alert, AlertVariant, Button } from '@patternfly/react-core';
import { format } from 'date-fns';
import { RequiredFieldsWarning } from '../required_fields_warning/required_fields_warning';
import { HealthStatus } from '../../schemas/cluster_status';
import { useLocation } from 'react-router';
import { HashLink } from 'react-router-hash-link';
interface LaunchAlertBannerProps {
  engagement: Engagement;
  onLaunch: (engagement: Engagement) => void;
  isLaunchable: boolean;
  requiredFields: string[];
  missingRequiredFields: string[];
}
const ENGAGEMENT_CARDS = {
  engagement_summary_card: 'Engagement Summary',
  poc_card: 'Point of Contact',
  oc_summary_card: 'Hosting Environment',
  user_card: 'Users',
};
const ENGAGEMENT_FIELD_MAP = {
  description: 'engagement_summary_card',
  location: 'engagement_summary_card',
  start_date: 'engagement_summary_card',
  end_date: 'engagement_summary_card',
  launch: 'engagement_summary_card',
  archive_date: 'engagement_summary_card',
  engagement_users: 'user_card',
  engagement_lead_name: 'poc_card',
  engagement_lead_email: 'poc_card',
  technical_lead_name: 'poc_card',
  technical_lead_email: 'poc_card',
  customer_contact_name: 'poc_card',
  customer_contact_email: 'poc_card',
  ocp_cloud_provider_name: 'oc_summary_card',
  ocp_cloud_provider_region: 'oc_summary_card',
  ocp_version: 'oc_summary_card',
  ocp_sub_domain: 'oc_summary_card',
  ocp_persistent_storage_size: { hash: 'oc_summary_card' },
  ocp_cluster_size: 'oc_summary_card',
  suggested_subdomain: 'oc_summary_card',
};
export function LaunchAlertBanner({
  engagement,
  onLaunch,
  isLaunchable,
  requiredFields,
  missingRequiredFields,
}: LaunchAlertBannerProps) {
  const overallStatus = engagement?.status?.overall_status;
  const path = useLocation().pathname;
  return (
    <Alert
      isInline
      title={
        engagement?.launch ? 'Engagement Launched' : 'Engagement Not Launched'
      }
      variant={statusAlert(overallStatus)}
      actionLinks={
        !engagement?.launch ? (
          <div>
            <div>
              <Button
                isDisabled={!isLaunchable}
                onClick={() => onLaunch(engagement)}
                data-cy={'launch_button'}
              >
                Launch
              </Button>
              <span style={{ margin: '0 0.5rem' }}>
                <RequiredFieldsWarning
                  missingRequiredFields={missingRequiredFields}
                  requiredFields={requiredFields}
                />
              </span>
            </div>
            <div style={{ margin: '0.5rem 0' }}>
              <span>
                Required fields are missing in the following sections:
              </span>
              <ul>
                {Array.from(
                  new Set(
                    missingRequiredFields.map(
                      field => ENGAGEMENT_FIELD_MAP[field]
                    )
                  )
                ).map(section => (
                  <li>
                    <HashLink smooth to={`${path}#${section}`}>
                      {ENGAGEMENT_CARDS[section]}
                    </HashLink>
                  </li>
                ))}
              </ul>
            </div>
          </div>
        ) : (
          undefined
        )
      }
    >
      <LaunchMessage engagement={engagement} />
    </Alert>
  );
}

interface LaunchMessageProps {
  engagement: Engagement;
}
export function LaunchMessage({ engagement }: LaunchMessageProps) {
  let launchMessage = '';
  if (!engagement?.launch) {
    launchMessage = 'This engagement has not been launched';
  } else {
    launchMessage = `This engagement was launched on ${format(
      engagement?.launch?.launched_date_time,
      'MMM d, yyyy'
    )} by ${engagement?.launch?.launched_by}`;
  }
  return <span>{launchMessage}</span>;
}

const statusAlert = (overallStatus?: HealthStatus) => {
  switch (overallStatus) {
    case HealthStatus.green: {
      return AlertVariant.success;
    }
    case HealthStatus.yellow: {
      return AlertVariant.warning;
    }
    case HealthStatus.red: {
      return AlertVariant.danger;
    }
    default: {
      return AlertVariant.info;
    }
  }
};
