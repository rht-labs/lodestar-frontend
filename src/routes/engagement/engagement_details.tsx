import React, { useEffect } from 'react';
import { Engagement } from '../../schemas/engagement';
import { useEngagement } from '../../context/engagement_context/engagement_hook';
import { Route, useParams, Switch, useRouteMatch } from 'react-router';
import { getValidatorsFromEngagementFormConfig } from '../../common/config_validator_adapter';
import {
  ValidationProvider,
  FormValidator,
} from '../../context/validation_context/validation_context';
import { EngagementDetailsViewTemplate } from '../../layout/engagement_details_view';
import { EngagementOverview } from './overview';
import { EngagementJsonDump } from './json_dump';
import { EngagementJsonSerializer } from '../../serializers/engagement/engagement_json_serializer';

export interface EngagementViewProps {
  currentEngagement?: Engagement;
}

export interface EngagementDetailViewProps {
  error: Error;
  currentEngagement: Engagement;
  createEngagementPoll: (
    engagement: Engagement
  ) => Promise<{ cancel: () => void }>;
  getEngagement: (
    customer_name: string,
    project_name: string
  ) => Promise<Engagement>;
}
export const validatorsWithDateValidators = (
  validators: FormValidator,
  dates: Pick<Engagement, 'start_date' | 'end_date'>
): FormValidator => {
  return {
    ...validators,
    end_date: [
      ...(validators['end_date'] ?? []),
      endDate => {
        const result =
          dates?.start_date > endDate
            ? 'End date must be after the start date'
            : null;
        return result;
      },
    ],
    archive_date: [
      ...(validators['archive_date'] ?? []),
      archiveDate => {
        const result =
          archiveDate < dates?.end_date
            ? 'Archive date must be after the end date'
            : null;
        return result;
      },
    ],
  };
};

export const EngagementDetailView = () => {
  const { project_name, customer_name } = useParams<{
    customer_name: string;
    project_name: string;
  }>();

  const {
    createEngagementPoll,
    saveEngagement,
    currentEngagement,
    currentChanges,
    engagementFormConfig,
  } = useEngagement({ projectName: project_name, customerName: customer_name });

  useEffect(() => {
    let engagementPoll;
    if (currentEngagement?.project_name && currentEngagement?.customer_name) {
      engagementPoll = createEngagementPoll(currentEngagement);
    }
    return async () => {
      (await engagementPoll)?.cancel?.();
    };
  }, [currentEngagement, createEngagementPoll]);

  const validators = getValidatorsFromEngagementFormConfig(
    engagementFormConfig
  );
  const { url } = useRouteMatch();
  const serializer = new EngagementJsonSerializer();
  const {
    commits,
    uuid,
    project_id,
    mongo_id,
    creation_details,
    status,
    launch,
    ...editableFields
  } = currentEngagement ?? {};
  const { end_date, start_date } = currentChanges;

  return (
    <ValidationProvider
      validators={validatorsWithDateValidators(validators, {
        end_date,
        start_date,
      })}
    >
      <EngagementDetailsViewTemplate
        engagement={currentEngagement}
        onSave={saveEngagement}
      >
        <Switch>
          <Route path={`${url}/json`}>
            <EngagementJsonDump
              json={JSON.stringify(
                serializer.serialize((editableFields ?? {}) as Engagement),
                null,
                2
              )}
              onSave={value => {
                saveEngagement(serializer.deserialize(JSON.parse(value)));
              }}
            />
          </Route>
          <Route path={`${url}/`}>
            <EngagementOverview />
          </Route>
        </Switch>
      </EngagementDetailsViewTemplate>
    </ValidationProvider>
  );
};
