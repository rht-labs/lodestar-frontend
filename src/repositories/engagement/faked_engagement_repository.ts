import { EngagementRepository } from './engagement_repository';
import { Engagement } from '../../models/engagement';
import faker from 'faker';

export class FakedEngagementRepository extends EngagementRepository {
  async fetchEngagements(): Promise<Engagement[]> {
    return new Array(8)
      .fill(null)
      .map(() => new Engagement({ name: faker.company.companyName() }));
  }
}
