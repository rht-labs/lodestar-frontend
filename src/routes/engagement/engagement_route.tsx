import React from 'react';
import { useAnalytics } from '../../context/analytics_context/analytics_context';
import { useSession } from '../../context/auth_context/auth_context';
import { useEngagementFormConfig } from '../../context/engagement_config_context/engagement_config_hook';
import { EngagementProvider } from '../../context/engagement_context/engagement_context';
import { useFeedback } from '../../context/feedback_context/feedback_context';
import { useServiceProviders } from '../../context/service_provider_context/service_provider_context';
import ScrollToTop from '../scroll_to_top';
import { EngagementDetailView } from './engagement_details';

export const EngagementRoute = () => {
  const { engagementService, categoryService } = useServiceProviders();
  const feedback = useFeedback();
  const analyticsContext = useAnalytics();
  const authContext = useSession();
  const { engagementFormConfig } = useEngagementFormConfig(engagementService);
  return (
    <EngagementProvider
      authContext={authContext}
      feedbackContext={feedback}
      analyticsContext={analyticsContext}
      categoryService={categoryService}
      engagementService={engagementService}
      engagementFormConfig={engagementFormConfig}
    >
      <ScrollToTop>
        <EngagementDetailView />
      </ScrollToTop>
    </EngagementProvider>
  );
};
