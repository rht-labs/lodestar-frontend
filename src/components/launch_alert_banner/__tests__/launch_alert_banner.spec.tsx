import React from 'react';
import { render } from '@testing-library/react';
import { LaunchAlertBanner } from '../launch_alert_banner';
import { Engagement } from '../../../schemas/engagement';
import { MemoryRouter } from 'react-router';
describe('Engagement Navigation ', () => {
  test('matches the snapshot', () => {
    expect(
      render(
        <MemoryRouter>
          <LaunchAlertBanner
            isLaunchable
            engagement={Engagement.fromFake(true)}
            missingRequiredFields={[]}
            onLaunch={() => {}}
            requiredFields={[]}
          />
        </MemoryRouter>
      )
    ).toMatchSnapshot();
  });
});
