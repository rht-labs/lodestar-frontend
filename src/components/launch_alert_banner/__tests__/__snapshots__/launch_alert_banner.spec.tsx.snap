// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement Navigation  matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-label="Success Alert"
        class="pf-c-alert pf-m-inline pf-m-success"
        data-ouia-component-id="0"
        data-ouia-component-type="PF4/Alert"
        data-ouia-safe="true"
      >
        <div
          class="pf-c-alert__icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 0 512 512"
            width="1em"
          >
            <path
              d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
              transform=""
            />
          </svg>
        </div>
        <h4
          class="pf-c-alert__title"
        >
          <span
            class="pf-u-screen-reader"
          >
            Success alert:
          </span>
          Engagement Not Launched
        </h4>
        <div
          class="pf-c-alert__description"
        >
          <span>
            This engagement has not been launched
          </span>
        </div>
        <div
          class="pf-c-alert__action-group"
        >
          <div>
            <div>
              <button
                aria-disabled="false"
                class="pf-c-button pf-m-primary"
                data-cy="launch_button"
                data-ouia-component-id="1"
                data-ouia-component-type="PF4/Button"
                data-ouia-safe="true"
                type="button"
              >
                Launch
              </button>
              <span
                style="margin: 0px 0.5rem;"
              >
                <svg
                  aria-describedby="pf-tooltip-1"
                  aria-hidden="true"
                  fill="green"
                  height="1em"
                  role="img"
                  style="vertical-align: -0.125em;"
                  viewBox="0 0 512 512"
                  width="1em"
                >
                  <path
                    d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                    transform=""
                  />
                </svg>
              </span>
            </div>
            <div
              style="margin: 0.5rem 0px;"
            >
              <span>
                Required fields are missing in the following sections:
              </span>
              <ul />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-label="Success Alert"
      class="pf-c-alert pf-m-inline pf-m-success"
      data-ouia-component-id="0"
      data-ouia-component-type="PF4/Alert"
      data-ouia-safe="true"
    >
      <div
        class="pf-c-alert__icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="1em"
          role="img"
          style="vertical-align: -0.125em;"
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            transform=""
          />
        </svg>
      </div>
      <h4
        class="pf-c-alert__title"
      >
        <span
          class="pf-u-screen-reader"
        >
          Success alert:
        </span>
        Engagement Not Launched
      </h4>
      <div
        class="pf-c-alert__description"
      >
        <span>
          This engagement has not been launched
        </span>
      </div>
      <div
        class="pf-c-alert__action-group"
      >
        <div>
          <div>
            <button
              aria-disabled="false"
              class="pf-c-button pf-m-primary"
              data-cy="launch_button"
              data-ouia-component-id="1"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe="true"
              type="button"
            >
              Launch
            </button>
            <span
              style="margin: 0px 0.5rem;"
            >
              <svg
                aria-describedby="pf-tooltip-1"
                aria-hidden="true"
                fill="green"
                height="1em"
                role="img"
                style="vertical-align: -0.125em;"
                viewBox="0 0 512 512"
                width="1em"
              >
                <path
                  d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
                  transform=""
                />
              </svg>
            </span>
          </div>
          <div
            style="margin: 0.5rem 0px;"
          >
            <span>
              Required fields are missing in the following sections:
            </span>
            <ul />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
